src/diagnostics/MOM_diagnostics.F90

```diff
@@ -257,6 +264,14 @@ subroutine calculate_diagnostic_fields(u, v, h, uh, vh, tv, ADp, CDp, p_surf, &
 
   call calculate_derivs(dt, G, CS)
 
+  ! Diagnostics for terms multiplied by fractional thicknesses
+  do j=js,je ; do I=Isq,Ieq
+    hf_du_dt_2d(I,j) = 0.0
+  enddo ; enddo
+  do J=Jsq,Jeq ; do i=is,ie
+    hf_dv_dt_2d(i,J) = 0.0
+  enddo ; enddo
+
```

Comment:
Most of the time explicit loops are preferred, but for initializing arrays like this, the style guide says you could have done `hf_du_dt_2d(:,:)=0.`

Reply:
Thanks. This is much shorter. I will use this form in all arrays.